src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -463,85 +467,127 @@ private Type EmitIListCoercion<T>(AssociativeNode arg)
-                        if (mbs == null || !mbs.Any())
-                        {
-                            mbs = type.GetMethods(BindingFlags.Public | BindingFlags.Instance).Where(
-                                m => m.Name == methodName && m.GetParameters().Length + 1 == args.Count).ToList();
-                        }
-
-                        // Check for property getters
-                        if (mbs == null || !mbs.Any())
-                        {
-                            var prop = type
-                                .GetProperties(BindingFlags.Public | BindingFlags.Static | BindingFlags.Instance).Where(
-                                    p => p.Name == methodName).FirstOrDefault();
-
-                            if (prop != null)
-                            {
-                                mbs = prop.GetAccessors().ToList();
-                            }
-                        }
-
-                        // Check for constructorinfo objects
-                        if(mbs == null || !mbs.Any())
-                        {
-                            mbs = type.GetConstructors().Where(
-                                m => m.DeclaringType.Name == methodName && m.GetParameters().Length == args.Count).ToList();
-                        }
-
-                        if (mbs != null && mbs.Any())
-                        {
-                            methodCache.Add(key, mbs);
-                            break;
-                        }
+                    methodName = ProtoCore.DSASM.Constants.kGetterPrefix + methodName;
+                    fg = runtimeCore.GetFuncGroup(methodName, className);
+                }
 
-                        //if (method != null)
-                        //{
-                        //    argTypes = method.GetParameters().Select(p => p.ParameterType).ToList();
-                        //    return method.ReturnType;
-                        //}
-                    }
+                // Try to find a setter
+                if (fg == null)
+                {
+                    methodName = ProtoCore.DSASM.Constants.kSetterPrefix + methodName;
+                    fg = runtimeCore.GetFuncGroup(methodName, className);
                 }
-                else
+            }
+
+            Validity.Assert(fg != null, "Did not find a function group");
+
+            foreach (var funcEnd in fg.FunctionEndPoints)
+            {
+                var procNode = funcEnd.procedureNode;
+                Validity.Assert(procNode != null, "Expected to have a valid procedureNode");
+
+                bool isStaticOrConstructorOrInternal = procNode.IsStatic || procNode.IsConstructor || CoreUtils.IsInternalMethod(methodName);
+
+                bool matchingArgCount = isStaticOrConstructorOrInternal ?
+                    procNode.ArgumentTypes.Count == args.Count :
+                    (procNode.ArgumentTypes.Count + 1) == args.Count;
+
+                if (!matchingArgCount) {
```

Comment:
In which cases will `matchingArgCount `be `false`?

Reply:
![image](https://user-images.githubusercontent.com/46732933/193310961-78e19a1c-3deb-477e-a52b-d4851ec735ed.png)